clients/src/main/java/org/apache/kafka/common/requests/SyncGroupRequest.java

```diff
@@ -81,6 +81,10 @@ public AbstractResponse getErrorResponse(int throttleTimeMs, Throwable e) {
         return groupAssignments;
     }
 
+    public boolean areProtocolTypeAndNamePresent() {
+        return version() >= 5 && (data.protocolType() == null || data.protocolName() == null);
```

Comment:
The version check is probably not necessary. Also, shouldn't we be checking for non-null entries?

Reply:
I think that we need the version check. `ProtocolType` and `ProtocolName` are null by default thus they will be null here for all the versions oldest than version 5 and we must accept them, right? However, for newer version, we want to ensure that both fields are provided.

Regarding the non-null entries, the verification (== provided values match the group's ones) is done in the `GroupCoordinator` when the fields are provided. Does it answer your question?

Reply:
Maybe I'm missing something, but the function name suggests that it should return true if both the protocol type and name are non-null. The current logic returns true if either field is null. Are we missing a `!`?

Reply:
Oh... The name is indeed misleading. I have reworked this part a bit. Now, the function is called `areMandatoryProtocolTypeAndNamePresent`. It returns true for any version above 5, and verifies that both fields are non-null for versions from 5 and above. The call is the Api layer has been updated accordingly. Apparently, my unit tests were broken as well as I was not calling the right function. Good catch!